core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/AmbariRepositoryVersionService.java

```diff
@@ -146,49 +132,29 @@ private void addRepository(StackService client, String stack, String version, St
         client.addStackRepository(stack, version, os, repoId, repoUrl, verify);
     }
 
-    private void addVersionDefinitionFileToAmbari(Long stackId, StackService ambariClient, StackRepoDetails stackRepoDetails) {
+    private void addVersionDefinitionFileToAmbari(StackService ambariClient, StackRepoDetails stackRepoDetails) {
         Optional<String> vdfUrl = Optional.ofNullable(stackRepoDetails.getStack().get(CUSTOM_VDF_REPO_KEY));
         if (!vdfUrl.isPresent()) {
-            vdfUrl = getVDFUrlByOsType(stackId, stackRepoDetails);
+            String message = "Couldn't determine any VDF file";
```

Comment:
I would add the cluster/stack name and id to the message to get a human readable exception message.